src/cpp/session/modules/SessionRSConnect.R

```diff
@@ -13,6 +13,24 @@
 #
 #
 
+.rs.addJsonRpcHandler("get_deployment_env_vars", function()
+{
+   # Find active .Renviron file
+   environFile <- .rs.findEnvironFile()
+   if (!nzchar(environFile) || !file.exists(environFile))
+      return(character())
+   
+   # Read environment variable names from the file
+   contents <- readLines(environFile, warn = FALSE)
+   pattern <- "^\\s*([\\w_]+)\\s*="
+   matchedLines <- grep(pattern, contents, perl = TRUE, value = TRUE)
+   envVars <- gsub("\\s*=.*", "", matchedLines)
+   
+   # Keep only those tokens which would likely have secrets of interest
+   pattern <- "(?:^|[\\b_])(?:api|key|pass|pat|secret|token)(?:$|[\\b_])"
```

Comment:
For users who have variables that don't match this pattern (you just know that's going to happen) should we make this configurable?

Perhaps a second regular expression (in a setting, initially empty) that specifies additional things to match? Of course, then we end up helping people debug regular expressions, so instead could be an allowlist where additional variables to match can be added.

Also, the other way around, variables that match this built-in pattern that they don't want to show up as a choice; in that case, either let them override the primary regex or maybe a blocklist.

All of this for future consideration, probably more of a server/workbench thing. I don't think we need to over think this for v1.